core/src/main/java/org/elasticsearch/rest/action/RestFieldStatsAction.java

```diff
@@ -98,7 +98,7 @@ public RestResponse buildResponse(FieldStatsResponse response, XContentBuilder b
                     builder.endObject();
                 }
                 builder.endObject();
-                return new BytesRestResponse(RestStatus.OK, builder);
+                return new BytesRestResponse(response.getStatus(), builder);
```

Comment:
Sure, though I think it makes it more consistent to change these to be the same, I'll open a separate PR for it after this one to change them to be 500 if a shard fails and remove all of these except for the refresh action